### PR TITLE
NIFI-9200: Free cache on heap after disabling AbstractCSVLookupService

### PR DESCRIPTION
**Pull request by Lehel44**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

CSVRecordLookupService and SimpleCsvFileLookupService use a Map cache that is not cleaned up after disabling the service, thus it remains on the heap.

https://issues.apache.org/jira/browse/NIFI-9200

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by simonbence — COMMENTED** [review]
LGTM, however, I would think adding a new test case or extending an existing one to check if the cache is being flushed would be beneficial.

File: nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/CSVRecordLookupService.java [hunk 1]
```diff
@@ -16,11 +16,13 @@
  */
 package org.apache.nifi.lookup;
 
+import avro.shaded.com.google.common.annotations.VisibleForTesting;
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/CSVRecordLookupService.java:19]: The non-shaded version should be used instead.

File: nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java [hunk 2]
```diff
@@ -111,5 +111,21 @@ public void testCsvRecordLookupServiceWithCustomSeparatorQuotedEscaped() throws
         assertEquals("my_value with an escaped |.", my_key.get().getAsString("value"));
     }
 
+    @Test
+    public void testCacheIsClearedWhenDisableService() throws InitializationException {
+        final TestRunner runner = TestRunners.newTestRunner(TestProcessor.class);
```
Comment (simonbence) [inline nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java:116]: Thanks for adding the tests, but please consider the following approach:
Step 1: Using the service in order to have cache content. 
Step 2: Check for the cache record by asking for it. The service should return the cached entry.
Step 3: Disable and enable the service.
Step 4: Check for the cache record by asking for it (again). The service should _not_ return any entry

I think this approach is more subtle and describes the case more. That decision, that the cache is not emptied for example but thrown away by assigning null to the attribute is something internal to the implementation.

**Review by turcsanyip — APPROVED** [review]
+1 LGTM Tested with a simple flow.

@Lehel44 Thanks for fixing it.
@simonbence Thanks for the review.

Merging to main.

**Pull request closed without merging**